packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -193,6 +193,27 @@ exports.createContentTypeNodes = ({
       // Get localized fields.
       const entryItemFields = _.mapValues(entryItem.fields, v => getField(v))
 
+      // If entry is not set by user, provide an empty value of the same type
+      contentTypeItem.fields.forEach(contentTypeItemField => {
+        const fieldName = contentTypeItemField.id
+        if (typeof entryItemFields[fieldName] === "undefined") {
+          if (contentTypeItemField.type.match(/Symbol|Text|Date/)) {
+            entryItemFields[fieldName] = ""
+          } else if (contentTypeItemField.type.match(/Number/)) {
+            // NOTE: Might be a problem for some people to auto set to 0?
+            entryItemFields[fieldName] = 0
```

Comment:
since `Number(undefined) === NaN` let's set the default to `NaN`

Reply:
👍

Reply:
good point. Thanks